File: power-adapters-sample/src/main/java/com/nextfaze/poweradapters/sample/CatView.java
package com.nextfaze.poweradapters.sample;

import android.content.Context;
import android.support.v7.widget.CardView;
import android.util.AttributeSet;
import android.widget.ImageView;
import android.widget.TextView;
import butterknife.BindView;
import butterknife.ButterKnife;
import com.squareup.picasso.Picasso;
import lombok.NonNull;

public final class CatView extends CardView {

    @BindView(R.id.image)
    ImageView mImageView;

    @BindView(R.id.title)
    TextView mTitleView;

    @BindView(R.id.subtitle)
    TextView mSubtitleView;

    public CatView(Context context) {
        this(context, null);
    }

    public CatView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public CatView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        setUseCompatPadding(true);
        inflate(context, R.layout.cat_view, this);
        ButterKnife.bind(this);
    }

    public void setCat(@NonNull Cat cat) {
        // Image
        Picasso.with(getContext())
                .load(cat.getImageUri())
                .into(mImageView);

        // Title
        mTitleView.setText(cat.getName());

        // Subtitle
        mSubtitleView.setText(cat.getCountry());
    }
}

File: power-adapters-sample/src/main/java/com/nextfaze/poweradapters/sample/cats/CatView.kt
package com.nextfaze.poweradapters.sample.cats

import android.content.Context
import android.net.Uri
import android.util.AttributeSet
import androidx.cardview.widget.CardView
import com.nextfaze.poweradapters.sample.R
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.cat_view.view.*
import kotlin.properties.Delegates.observable

class CatView @JvmOverloads constructor(
        context: Context,
        attrs: AttributeSet? = null,
        defStyleAttr: Int = 0
) : CardView(context, attrs, defStyleAttr) {

    var cat by observable<Cat?>(null) { _, _, cat ->
        if (cat != null) {
            Picasso.with(context).load(generateImageUri(cat.name)).into(imageView)
            titleView.text = cat.name
            subtitleView.text = cat.country
        }
    }

    init {
        inflate(context, R.layout.cat_view, this)
        useCompatPadding = true
    }
}

private fun generateImageUri(catName: String) =
        Uri.parse("http://thecatapi.com/api/images/get?api_key=MjM0NTc2").buildUpon().appendQueryParameter("cat", catName).build()
